# 🧼 Data Cleaning & Feature Engineering

## 📌 Objective
The goal of this notebook is to clean the raw Netflix dataset and prepare it for further analysis by handling data types, missing values, and extracting useful features.

---

## 📂 Dataset Overview
The dataset contains information about Netflix Originals including:
- Title
- Genre
- Premiere Date
- Runtime
- IMDB Score
- Language

In [1]:
import pandas as pd

df = pd.read_csv("../data/netflix_content_dataset.csv", encoding="ISO-8859-1")

df.head()
df.info()
df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 584 entries, 0 to 583
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       584 non-null    object 
 1   Genre       584 non-null    object 
 2   Premiere    584 non-null    object 
 3   Runtime     584 non-null    int64  
 4   IMDB Score  584 non-null    float64
 5   Language    584 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 27.5+ KB


Title         0
Genre         0
Premiere      0
Runtime       0
IMDB Score    0
Language      0
dtype: int64

## 🔍 Initial Data Inspection

- Checked dataset structure using `df.info()`
- Verified missing values using `df.isnull().sum()`
- Observed that:
  - No missing values in key columns
  - 'Premiere' column is in object format
  - 'Runtime' needs cleaning

In [11]:
# Fix date
df['Premiere'] = pd.to_datetime(df['Premiere'], errors='coerce')
df = df.dropna(subset=['Premiere'])

# Extract features
df['release_year'] = df['Premiere'].dt.year
df['month'] = df['Premiere'].dt.month_name()

# Clean runtime safely
df['duration_min'] = (
    df['Runtime']
    .astype(str)
    .str.replace(' min','', regex=False)
    .str.replace('min','', regex=False)
)

df['duration_min'] = pd.to_numeric(df['duration_min'], errors='coerce')
df = df.dropna(subset=['duration_min'])

print("Dataset Shape:", df.shape)
df.head()

Dataset Shape: (579, 9)


,Title,Genre,Premiere,Runtime,IMDB Score,Language,release_year,month,duration_min
0,Enter the Anime,Documentary,2019-08-05,58,2.5,English/Japanese,2019,August,58
1,Dark Forces,Thriller,2020-08-21,81,2.6,Spanish,2020,August,81
2,The App,Science fiction/Drama,2019-12-26,79,2.6,Italian,2019,December,79
3,The Open House,Horror thriller,2018-01-19,94,3.2,English,2018,January,94
4,Kaali Khuhi,Mystery,2020-10-30,90,3.4,Hindi,2020,October,90


## ⚙️ Feature Engineering

To make the dataset more useful for analysis, the following transformations were applied:

### ✔ Date Conversion
- Converted 'Premiere' column to datetime format
- Handled invalid values using `errors='coerce'`

### ✔ New Features Created
- `release_year` → Extracted year from premiere date
- `month` → Extracted month name

### ✔ Runtime Cleaning
- Removed 'min' text from runtime column
- Converted runtime into numeric format (`duration_min`)

### 📏 Dataset Size
The dataset contains rows and columns after cleaning, ensuring data is ready for analysis.

### 🔍 Insight:
After cleaning, the dataset contains 579 records and 9 features, ensuring high data quality for analysis.

In [12]:
df.columns

Index(['Title', 'Genre', 'Premiere', 'Runtime', 'IMDB Score', 'Language',
       'release_year', 'month', 'duration_min'],
      dtype='object')

### 🧾 Columns in Dataset
The dataset includes important features such as title, genre, release date, runtime, IMDB score, and language.

In [7]:
df.to_csv("../data/cleaned_netflix_data.csv", index=False)

## 💾 Saving Cleaned Data

The cleaned dataset is saved for further analysis in the next notebook.

## 📊 Final Cleaned Data

- Dataset successfully cleaned and transformed
- No missing values in critical columns
- New features added for better analysis

### ✅ Key Improvements:
- Standardized date format
- Converted runtime to numeric
- Extracted time-based features

This cleaned dataset is now ready for Exploratory Data Analysis (EDA).

## 🚀 Next Step

The cleaned dataset will now be used for Exploratory Data Analysis (EDA) to uncover trends and patterns in Netflix content.